azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,447 @@
+						"minute": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"time_zone_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureVirtualMachineTimeZone(),
+			},
+
+			"notification_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"status": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  dtl.NotificationStatusDisabled,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(dtl.NotificationStatusEnabled),
+								string(dtl.NotificationStatusDisabled),
+							}, false),
+						},
+						"time_in_minutes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"webhook_url": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDevTestLabSchedulesCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).devTestLabs.LabSchedulesClient
+
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	devTestLabName := d.Get("lab_name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, devTestLabName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Schedule %q (Dev Test Lab %q / Resource Group %q): %s", name, devTestLabName, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_dev_test_schedule", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	tags := d.Get("tags").(map[string]interface{})
+
+	schedule := dtl.Schedule{
+		Location:           &location,
+		ScheduleProperties: &dtl.ScheduleProperties{},
+		Tags:               expandTags(tags),
+	}
+
+	switch status := d.Get("status"); status {
+	case string(dtl.EnableStatusEnabled):
+		schedule.ScheduleProperties.Status = dtl.EnableStatusEnabled
+	case string(dtl.EnableStatusDisabled):
+		schedule.ScheduleProperties.Status = dtl.EnableStatusDisabled
+	default:
+	}
+
+	if taskType := d.Get("task_type").(string); taskType != "" {
+		schedule.ScheduleProperties.TaskType = &taskType
+	}
+
+	if timeZoneId := d.Get("time_zone_id").(string); timeZoneId != "" {
+		schedule.ScheduleProperties.TimeZoneID = &timeZoneId
+	}
+
+	if _, ok := d.GetOk("weekly_recurrence"); ok {
+		weekRecurrence := expandArmDevTestLabScheduleRecurrenceWeekly(d)
+
+		schedule.WeeklyRecurrence = weekRecurrence
+	}
+
+	if _, ok := d.GetOk("daily_recurrence"); ok {
+		dailyRecurrence := expandArmDevTestLabScheduleRecurrenceDaily(d)
+		schedule.DailyRecurrence = dailyRecurrence
```

Comment:
This could easily be:
```suggestion
		schedule.DailyRecurrence = expandArmDevTestLabScheduleRecurrenceDaily(v.(type))
```

Reply:
Done